packages/text-input/src/TextInput/TextInput.tsx

```diff
@@ -198,9 +200,9 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
               inputStateStyles[state][theme],
               inputFocusStyles[theme], // Always show focus styles
               {
-                [css`
-                  padding-right: 60px;
-                `]: optional && !disabled,
+                [inputPaddingForIndicator[sizeVariant]]: state !== State.None,
+                [inputPaddingForOptionalText[sizeVariant]]:
+                  state === State.None && optional,
```

Comment:
Should we also add `!disabled`? the padding remains when disabled and optional which leaves a large gap on the right.
![Screenshot 2023-04-27 at 11 39 48 AM](https://user-images.githubusercontent.com/5740159/234914236-5127d790-f483-4137-a67b-57570be39599.png)